src/Datadog.Trace/Tracer.cs

```diff
@@ -212,15 +212,20 @@ public static Tracer Instance
         {
             get
             {
-                return LazyInitializer.EnsureInitialized(ref _instance, ref _globalInstanceInitialized, ref _globalInstanceLock);
+                _globalManualReset.Wait();
+
+                lock (_globalInstanceLock)
+                {
+                    return _instance;
+                }
             }
 
             set
```

Comment:
can we get rid of the setter?

Reply:
We have `AspNetCoreDiagnosticObserverTests` and `CorrelationIdentifierTests` that are using it. Maybe even more, but these are the first ones that pop up.